packages/web/src/components/list/SingleDropdownList.js

```diff
@@ -249,6 +249,10 @@ class SingleDropdownList extends Component {
 		const { isLastBucket } = this.state;
 		let selectAll = [];
 
+		if (this.props.loading && this.props.loader) {
```

Comment:
Would it better if we create a helper util for this - say `renderLoader` which takes the `props` and renders the loader when needed?

Reply:
but still we have to write the condition to render the loader.
Can you please describe here how we can use a util function here & what'll be the advantages?

Reply:
Sure, we can import that util method in every component:

```js
import { renderLoader } from '...';

// inside component render:
render() {
    return (
        <div>
            {renderLoader(props)}
        </div>
    )
}
```

Now, we don't have to repeat the check for `props.isLoading && props.loader` everywhere - since the rendering of the loader is completely dependent on `props`.

Reply:
But that's not the case, we just need to render `loader` ( nothing else so we have to write a check before renderLoader ) if `isLoading` is `true`.

Reply:
Not sure if I am super clear on that. Here is what I have in mind:

```js
const renderLoader = (props) => {
    if (props.isLoading && props.loader) {
        return props.loader;
    }
    return null;
}
```

So we are checking `isLoading` & `loader` props here in every component right?

Reply:
that's ok, just want to do the same but the problem is that we still need to write the check to determine if `isLoading` true then only we have to render the loader.
So, the code will be like that with that snippet
```js
render() {
   if (isLoading) { return renderLoader(this.props); };
   // other code
}
```
which doesn't has any advantage

Reply:
um, we are checking for `isLoading` inside the `renderLoader` util, so why check it before calling the function?